cpp/tests/filter/helper/utils.cu

```diff
@@ -69,20 +61,27 @@ gdf_size_type count_zero_bits(gdf_valid_type *valid, size_t column_size)
     return numbits;
 }
 
-std::string chartobin(gdf_valid_type c, int size/* = 8*/)
-{
-    std::string bin;
-    bin.resize(size);
-    bin[0] = 0;
-    int i;
-    for (i = size - 1; i >= 0; i--)
-    {
-        bin[i] = (c % 2) + '0';
-        c /= 2;
-    }
-    return bin;
-}
-
 auto print_binary(gdf_valid_type n, int size) -> void {
     std::cout << chartobin(n) << "\t sz: " <<  size <<  "\tbinary: " << chartobin(n, size) << std::endl;
 }
+
+// Create the valid pointer and init randomly the last half column
+void initialize_valids(host_valid_pointer& valid_ptr, size_t length, bool all_bits_on)
```

Comment:
Can you use the existing test util functionality in tests/utilities/cudf_test_utils.h? e.g. `gdf_col_pointer init_gdf_column(std::vector<T> data, size_t col_index, valid_initializer_t bit_initializer)` takes a bit_initializer functor/lambda. Example in column_test.cu look for `bit_setter`.